src/devices_models/devices/renewable_generation.jl

```diff
@@ -106,18 +106,19 @@ end
 
 function activepower_constraints!(psi_container::PSIContainer,
                                 devices::IS.FlattenIteratorWrapper{R},
-                                device_formulation::Type{<:AbstractRenewableDispatchFormulation},
+                                model::DeviceModel{R, <:AbstractRenewableDispatchFormulation},
```

Comment:
Can this be `model::DeviceModel{R, AbstractRenewableDispatchFormulation}` instead?  Same comment for other functions that take a DeviceModel.

Reply:
removing the `<:` causes errors...